docs/operations/advanced/kubernetes-deployment.md

```diff
@@ -0,0 +1,620 @@
+cluster_name         = "kms-production-party1"
+namespace            = "kms-threshold"
+service_account_name = "mpc-party-1"
+create_namespace     = true
+
+# IRSA Configuration (recommended for production)
+create_irsa = true
+
+# Tagging
+owner = "mpc-ops-team"
+additional_tags = {
+  "Project"     = "kms-threshold"
+  "Environment" = "production"
+  "Party"       = "1"
+}
+
+# RDS Configuration (for KMS connector)
+enable_rds              = true
+rds_db_name             = "kmsconnector"
+rds_username            = "kmsconnector"
+rds_deletion_protection = true
+
+# Node Group Configuration
+create_nodegroup                         = true
+nodegroup_instance_types                 = ["c7a.16xlarge"]
+nodegroup_min_size                       = 1
+nodegroup_max_size                       = 3
+nodegroup_desired_size                   = 1
+nodegroup_disk_size                      = 100
+nodegroup_capacity_type                  = "ON_DEMAND"
+nodegroup_ami_type                       = "AL2023_x86_64_STANDARD"
+nodegroup_ami_release_version           = "1.32.3-20250620"
+
+nodegroup_labels = {
+  "nodepool"    = "kms"
+  "party-id"    = "1"
+  "environment" = "production"
+}
+
+# Nitro Enclaves Configuration
+nodegroup_enable_nitro_enclaves         = true
+nodegroup_enable_ssm_managed_instance   = true
+kms_enabled_nitro_enclaves              = true
+kms_image_attestation_sha               = "5292569b5945693afcde78e5a0045f4bf8c0a594d174baf1e6bccdf0e6338ebe46e89207054e0c48d0ec6deef80284ac"
+kms_deletion_window_in_days             = 30
+EOF
+
+# Repeat for parties 2, 3, 4 with appropriate party_id and unique names
+```
+
+### Step 3: Deploy Infrastructure
+
+Deploy each party in sequence:
+
+```bash
+# Deploy Party 1
+terraform init
+terraform plan -var-file="party1.tfvars"
+terraform apply -var-file="party1.tfvars"
+
+# Get cluster credentials
+aws eks update-kubeconfig --region eu-west-1 --name kms-production-party1
+
+# Verify deployment
+kubectl get nodes -l nodepool=kms
+kubectl get pods -n kms-threshold
+```
+
+### Step 4: Configure Inter-Party Networking
+
+After all parties are deployed, configure VPC endpoints for secure communication:
+
+```bash
+# Deploy VPC endpoint provider (on each party)
+cd ../mpc-network-provider
+terraform init
+terraform apply -var="cluster_name=kms-production-party1" \
+                -var="service_name=kms-threshold-service"
+
+# Deploy VPC endpoint consumer (on each party to connect to others)
+cd ../mpc-network-consumer
+terraform init
+terraform apply -var="cluster_name=kms-production-party1" \
+                -var="external_parties=[{name=\"party2\", service_name=\"...\"}]"
+```
+
+---
+
+## KMS Application Deployment
+
+### Step 1: Create KMS Configuration
+
+```yaml
+# kms-threshold-config.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kms-threshold-config
+  namespace: kms-threshold
+data:
+  threshold.toml: |
+    [service]
+    listen_address = "0.0.0.0"
+    listen_port = 50100
```

Comment:
I am not sure where, but I think some place we should mention that it is absolutely crucial that this port is not exposed to the internet.